reconcile/utils/saasherder/interfaces.py

```diff
@@ -380,10 +380,17 @@ class SaasRole(Protocol):
 SaasPipelinesProviders = Union[SaasPipelinesProviderTekton, SaasPipelinesProvider]
 
 
+@runtime_checkable
+class ManagedResourceName(Protocol):
+    resource: str
+    resource_names: list[str]
+
+
 class SaasFile(HasParameters, HasSecretParameters, Protocol):
     path: str
     name: str
     managed_resource_types: list[str]
+    managed_resource_names: Optional[list[ManagedResourceName]]
```

Comment:
this must be written as

```
    @property
    def managed_resource_names(self) -> Optional[list[ManagedResourceName]]:
        ...
```

Yes, it's stupid and I hate it too.

Reply:
https://github.com/app-sre/qontract-reconcile/pull/3814/commits/ae8ecec604dc05b1544f73f920a29598df19260d